### DISCUSSION TITLE
How do I run the tools after BP_CARGO_INSTALL_TOOLS?

### DISCUSSION THREAD

**Discussion by chcoomilk** [paketo-buildpacks/feedback#37] [Q&A]
I am currently trying to deploy a Rust application with Diesel. I successfully installed the diesel_cli tool after configuring the key: "BP_CARGO_INSTALL_TOOLS" and the value: "diesel_cli".

The build process basically printed
> [200] => Installed package `diesel_cli v2.0.0` (executable `diesel`)

But I have no idea where it is and how to run it.

**Comment by chcoomilk** [comment]
I'm sorry, but I'm really confused understanding this config in the docs

<html><body>


$BP_CARGO_INSTALL_TOOLS | Additional tools that should be installed by running cargo install. This should be a space separated list, and each item should contain the name of the tool to install like cargo-bloat or diesel_cli. Tools installed will be installed prior to compiling application source code and will be available on $PATH during build execution (but are not installed into the runtime container).
-- | --



</body>
</html>

Tools will be available on $PATH, so I am assuming I can call it explicitly; for example (say diesel_cli installs the binary file called: diesel) and the command would be `diesel migration run` or `diesel setup`, right? But because it only exists during build execution, it's not going to be called in a Procfile. So is it going to be called after `$BP_CARGO_INSTALL_TOOLS_ARGS`, it will probably look like this `cargo install diesel_cli --force --no-default-features --features postgres && diesel migration run` or not at all?

**Reply by sophiewigmore** [reply]
cc @paketo-buildpacks/rust-maintainers

**Reply by dmikusa** [reply]
So this is kind of a new feature and we're still working out how people might want to use it. Your feedback on this is important.

What I can say is that the original intent was to be able to use this to install CLI tools that you might need to call from [a `build.rs` file](https://doc.rust-lang.org/cargo/reference/build-scripts.html), or possibly call from another buildpack.

That's why we don't install the tools into the container at runtime. I can see where that might make sense for some tools though. You might want to run `diesel` to do a migration at runtime. 

If you want to open an issue on https://github.com/paketo-community/cargo/issues for adding tools to the runtime, we can certainly take a look.
